File: data-starter/src/main/java/de/hipp/pnp/DataServiceApplication.java
package de.hipp.pnp;

import de.hipp.pnp.data.language.LanguageKeyConfiguration;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication(scanBasePackages = "de.hipp.*")
public class DataServiceApplication {

	@Autowired
	LanguageKeyConfiguration config;

	public static void main(String[] args) {
		SpringApplication.run(DataServiceApplication.class, args);

	}
}

File: data-starter/src/main/kotlin/de/hipp/pnp/DataServiceApplication.kt
package de.hipp.pnp

import org.springframework.aot.hint.RuntimeHints
import org.springframework.aot.hint.RuntimeHintsRegistrar
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.ImportRuntimeHints

@SpringBootApplication(scanBasePackages = ["de.hipp.*"], proxyBeanMethods = false)
@ImportRuntimeHints(DataServiceRuntimeHints::class)
open class DataServiceApplication {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(DataServiceApplication::class.java, *args)
        }
    }
}

class DataServiceRuntimeHints : RuntimeHintsRegistrar {
    override fun registerHints(
        hints: RuntimeHints,
        classLoader: ClassLoader?,
    ) {
        // Register the companion class for reflection
        hints.reflection().registerType(DataServiceApplication.Companion::class.java)
    }
}
